test/functional/bulk_tests.js

```diff
@@ -937,6 +937,69 @@ describe('Bulk', function() {
     }
   });
 
+  it(
+    'should provide descriptive error message for Unordered Batch with duplicate key errors on updates',
+    {
+      metadata: { requires: { topology: ['single', 'replicaset', 'ssl', 'heap', 'wiredtiger'] } },
+
+      test: function(done) {
+        const configuration = this.configuration;
+        var client = configuration.newClient(configuration.writeConcernMax(), {
+          poolSize: 1
+        });
+
+        client.connect(function(err, client) {
+          var db = client.db(configuration.db);
+          var col = db.collection('err_batch_write_unordered_ops_legacy_6');
+
+          // Write concern
+          var writeConcern = configuration.writeConcernMax();
+          writeConcern.unique = true;
+          writeConcern.sparse = false;
+
+          // Add unique index on b field causing all updates to fail
+          col.ensureIndex({ b: 1 }, writeConcern, function(err) {
```

Comment:
Since `ensureIndex` is deprecated in favor of `createIndexes`, can we use `col.createIndexes` instead? Here are the [docs](https://docs.mongodb.com/manual/reference/command/createIndexes/) for how to use `createIndexes`.